src/std_extensions/arithmetic/int_types.rs

```diff
@@ -30,37 +29,22 @@ fn int_custom_type(n: u8) -> CustomType {
 /// Integer type of a given bit width.
 /// Depending on the operation, the semantic interpretation may be unsigned integer, signed integer
 /// or bit string.
-pub fn int_type(n: u8) -> Type {
-    Type::new_extension(int_custom_type(n))
+pub fn int_type(width_power: u8) -> Type {
+    Type::new_extension(int_custom_type(width_power))
 }
 
-fn is_valid_width(n: u8) -> bool {
-    (n == 1)
-        || (n == 2)
-        || (n == 4)
-        || (n == 8)
-        || (n == 16)
-        || (n == 32)
-        || (n == 64)
-        || (n == 128)
+const fn is_valid_width(n: u8) -> bool {
+    (n <= (1u8 << (POWERS_OF_TWO - 1))) && ((n & (n - 1)) == 0)
 }
 
+const POWERS_OF_TWO: usize = 8;
+
 /// Get the bit width of the specified integer type, or error if the width is not supported.
-pub fn get_width(arg: &TypeArg) -> Result<u8, SignatureError> {
-    let n: u8 = match arg {
-        TypeArg::USize(n) => *n as u8,
-        _ => {
-            return Err(TypeArgError::TypeMismatch {
-                arg: arg.clone(),
-                param: TypeParam::USize,
-            }
-            .into());
-        }
-    };
-    if !is_valid_width(n) {
-        return Err(TypeArgError::InvalidValue(arg.clone()).into());
+pub fn get_width_power(arg: &TypeArg) -> u8 {
```

Comment:
I am a bit confused by this `width_power` nomenclature. What this returns is an integer width, right? A widely-understood concept being the number of bits in the int representation. It can be seen as a power of the tag of the SimplePredicate, but I don't think that really helps - indeed that's a description of the *implementation* of this method, but the purpose of this method is to isolate the details of this encoding from the rest of the method, right? Once I've got it, it's a width, I'm not expecting to take logarithm or do anything fancy with it, right?

So, alternative suggestions for naming:
* `int_width_as_power_of_simple_predicate` clearly too long but maximally explicit
* `int_width_as_power` shortened form of previous
* `int_width_from_typearg`, `int_width_from_tag`, `int_width_from_simp_pred` / `int_width_from_sum` etc.
* or maybe just `int_width` or `get_width`...

Reply:
it should be `get_log_width`, I would like to use the log of the width everywhere

Reply:
Yes, actually that was my initial suspicion!! Lol.

However, I see that this is only used in tests; in prod code. it is *always* followed immediately by passing the returned value to `int_type`. So you could just have one function `int_type_from_type_arg` (or similar) that does both together and I think that would completely avoid the opportunity for confusion

Reply:
hmmm, if it's called `get_log_width` I think that'd be clear, ok. Note the checks in `int_ops.rs` (that `n>m` etc.) would be equally valid on the actual width, if that was easier (probably not). Moreover, `int_type` is actually performing the inverse operation to this, so this is just validating - i.e. panicking....

Reply:
changed so int_type is just constructed from the arg directly, and the checks are performed on the log width where necessary

Reply:
now fewer lines overall!